src/ser/serializer.rs

```diff
@@ -300,8 +300,12 @@ where
     }
 
     #[inline]
-    fn serialize_struct(self, _name: &'static str, _len: usize) -> Result<Self::SerializeStruct> {
-        Ok(self)
+    fn serialize_struct(self, name: &'static str, _len: usize) -> Result<Self::SerializeStruct> {
```

Comment:
This makes me a little worried about degrading performance for all other cases to optimize specifically for the byte array case.

Reply:
I believe (have not measured) that the performance degradation is going to be minimal. The extra code is a single pointer comparison, likely optimized out because this is in monomorphized code.

It's also the same way `serde_json::value::RawValue` is implemented. I believe `serde_json` is a high-performance serde JSON implementation ensured by benchmarks.